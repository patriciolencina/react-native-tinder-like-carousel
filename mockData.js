export const userJSONString =
  '{"results":[{"user":{"gender":"male","name":{"title":"mr","first":"reginald","last":"burton"},"location":{"street":"5668 andreas ave","city":"duncanville","state":"iowa","zip":"69310"},"email":"reginald.burton77@example.com","username":"beautifultiger643","password":"calgary","salt":"o50omLUi","md5":"59f4e2db81840e41eaa1b2d39f2ead55","sha1":"f1d65fd8f818137c5bc3f4bc147bb8ea109d5867","sha256":"b5a525ca0e449c81029829d7b7eca752fce49b696c58363b9e36cfd272f18e5f","registered":"1246053519","dob":"144374909","phone":"(712)-125-8909","cell":"(899)-446-4844","SSN":"707-51-3358","picture":"http://api.randomuser.me/portraits/men/61.jpg"},"seed":"11059646270402eb","version":"0.4"}]}';

export const userData = {
  gender: 'male',
  name: {
    title: 'mr',
    first: 'reginald',
    last: 'burton',
  },
  location: {
    street: '5668 andreas ave',
    city: 'duncanville',
    state: 'iowa',
    zip: '69310',
  },
  email: 'reginald.burton77@example.com',
  username: 'beautifultiger643',
  password: 'calgary',
  salt: 'o50omLUi',
  md5: '59f4e2db81840e41eaa1b2d39f2ead55',
  sha1: 'f1d65fd8f818137c5bc3f4bc147bb8ea109d5867',
  sha256: 'b5a525ca0e449c81029829d7b7eca752fce49b696c58363b9e36cfd272f18e5f',
  registered: '1246053519',
  dob: '144374909',
  phone: '(712)-125-8909',
  cell: '(899)-446-4844',
  SSN: '707-51-3358',
  picture: 'http://api.randomuser.me/portraits/men/61.jpg',
};
