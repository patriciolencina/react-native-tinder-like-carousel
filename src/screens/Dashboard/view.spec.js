import React from 'react';

import {shallow, ShallowWrapper} from 'enzyme';
import {SafeAreaView} from 'react-native';
import DashboardView from './view';
import {userData} from '../../../mockData';

const createTestProps = (props: Object) => ({
  navigation: {
    navigate: jest.fn(),
  },
  users: [userData],
  moveRight: jest.fn(),
  moveLeft: jest.fn(),
  ...props,
});

describe('DashboardView', () => {
  describe('rendering', () => {
    let wrapper: ShallowWrapper;
    let props: any; // use type "any" to opt-out of type-checking
    beforeEach(() => {
      props = createTestProps({});
      wrapper = shallow(<DashboardView {...props} />); // no compile-time error
    });

    it('should match to snapshot', () => {
      expect(shallow(<DashboardView />)).toMatchSnapshot();
    });

    it('should render a <SafeAreaView />', () => {
      expect(wrapper.find(SafeAreaView)).toHaveLength(1);
    });
  });
});
