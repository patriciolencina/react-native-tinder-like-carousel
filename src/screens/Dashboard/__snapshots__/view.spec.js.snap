// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardView rendering should match to snapshot 1`] = `
<SafeAreaView>
  <Text>
    Loading...
  </Text>
</SafeAreaView>
`;
